atlp-rwanda/rca-phantom-team1-bn
# chore: Log In

# Add login functionality for driver and operator roles

## Description

This pull request implements the login feature for all roles. It adds a login form with validation and error handling, authentication logic using JWT and bcrypt.

## How to test

- Run `npm install` to install the dependencies
- Run `npm run dev` to start the development server
- Navigate to `http://localhost:{PORT}/api-docs` for swagger documentation and head to Auth on auth/login endpoint.
-  Enter the credentials provided by the operator or administrator
- Verify that you can log in successfully and access the tasks assigned to your role
- Verify that you cannot log in with invalid credentials or access tasks that are not authorized for your role


## Screenshots

![image](https://user-images.githubusercontent.com/68426990/226558472-ad2e6713-a52e-41c2-8665-3ef91d856288.png)
![image](https://user-images.githubusercontent.com/68426990/226558372-5c32773a-a732-46d0-8765-8c150a10596e.png)

## Checklist

- [x] I have added comments and documentation where necessary
- [x] I have followed the coding standards and style guidelines
